pkg/lisafs/handlers.go

```diff
@@ -1315,6 +1316,16 @@ func RenameAtHandler(c *Connection, comm Communicator, payloadLen uint32) (uint3
 			newDir.node.insertChildLocked(newName, renamed)
 			newDir.node.childrenMu.Unlock()
 
+			if replaced != nil && flags&linux.RENAME_EXCHANGE != 0 {
+				replaced.parent.DecRef(nil)
+				replaced.parent = oldDir.node
+				replaced.parent.IncRef()
+				replaced.name = oldName
+				oldDir.node.childrenMu.Lock()
+				oldDir.node.insertChildLocked(oldName, replaced)
+				oldDir.node.childrenMu.Unlock()
+				notifyRenameRecursive(replaced)
```

Comment:
This code is unreachable as the outter if condition `flags&linux.RENAME_EXCHANGE == 0`. I think you want to do:
```
if replaced != nil {
    if flags&linux.RENAME_EXCHANGE == 0 {
        ...
    } else {
        ...
    }
}
```

Reply:
`if replaced != nil && flags&linux.RENAME_EXCHANGE == 0` is not an outer `if`, it just looks so because of the diff cut.
If you open lines 1304-1314 you will see it.